pkg/azuredisk/azdrivernodeplugin.go

```diff
@@ -0,0 +1,103 @@
+// +build azurediskv2
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredisk
+
+import (
+	"context"
+	"time"
+
+	//corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientSet "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	azuredisk "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1alpha1"
+	azVolumeClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+)
+
+
+var partitionLabel  = "azdrivernodes.disk.csi.azure.com/partition"
+
+// RegisterAzDriverNodeOrDie creates custom resource for this driverNode
+func RegisterAzDriverNodeOrDie(NodeID string,
+	k clientSet.Interface,
```

Comment:
Rather than pass these parameters in separately, consider using `DriverV2` as a receiver since this implementation is tied to the V2 driver object anyway.

Same with RunAzDriverNodeHeartbeatLoop.